### PR TITLE
Fix: Conflicting profile properties between profile and attributes

### PR DESCRIPTION
**Pull request by kriss1897**
# Description
https://github.com/node-saml/passport-saml/blob/f1a436fa243080f8cfcf06fd0f40ce3af396c8b6/src/passport-saml/saml.ts#L1319
It is currently possible to override some default properties of the `profile` object in `processValidlySignedAssertionAsync` method.
This can be done if any property like `Issuer` is also present in the attribute mapping. The expected behavior here should be that the properties (`Issuer`, `inResponseTo`, `sessionIndex`, `nameID`, `nameIDFormat`, `nameQualifier`, `spNameQualifier`) will probably already be present in the profile, and in that case they shouldn't be overridden by the same named attribute in attribute mapping.

This issue can be handled gracefully by simply skipping the value from attribute mapping or an error can be thrown. This PR handles the issue gracefully. 

# Checklist:

- Issue Addressed: [#543]
- Link to SAML spec: [ ]
- Tests included? Yes
- Documentation updated? Not Required

### CODE REVIEW

File: src/node-saml/saml.ts [hunk 1]
```diff
@@ -1175,11 +1175,20 @@ class SAML {
             // if attributes has no AttributeValue child, continue
             return;
           }
+          const name = attribute.$.Name;
           const value = attribute.AttributeValue;
+
+          // If any property is already present in profile and is also present
+          // in attributes, then skip the one from attributes. Handle this
+          // conflict gracefully without returning any error
+          if (Object.prototype.hasOwnProperty.call(profile, name)) {
```
Comment (cjbarth) [inline src/node-saml/saml.ts:1184]: Do you think there is value is putting all these values in a child object on the profile in addition to cloning them directly to the profile object where there is no conflict? This way, nothing is lost, but we also don't break existing behavior.

Comment (kriss1897) [reply src/node-saml/saml.ts:1184]: Yes. I think putting all attributes in separate child property would be useful. And only clone the properties that do not exist on profile.

Comment (kriss1897) [reply src/node-saml/saml.ts:1192]: Update PR with the requested changes

**Comment by cjbarth** [comment]
This looks good to me. However, either we need to make this PR only against the [node-saml](https://github.com/node-saml/node-saml) project, or we need to land this PR in both this and that project.

**Comment by cjbarth** [comment]
Sounds good to me. As soon as I see a corresponding PR in the [node-saml](https://github.com/node-saml/node-saml) project, we'll land it here.

**Comment by kriss1897** [comment]
@cjbarth Added same PR on node-saml repo. https://github.com/node-saml/node-saml/pull/5

**Review by cjbarth — APPROVED** [review]

**Pull request merged**